app/src/main/java/com/itssuryansh/taaveez/Notes.kt

```diff
@@ -60,20 +74,60 @@ class Notes : AppCompatActivity() {
 
         }
 
-
         val NotesDao = (application as NotesApp).db.NotesDao()
         binding?.idFABAdd?.setOnClickListener {
 //            NewPoemDialog(NotesDao)
             val intent = Intent(this@Notes, Add_New_Content::class.java)
             startActivity(intent)
+
         }
 
-        lifecycleScope.launch {
-            NotesDao.fetchAllNotes().collect {
-                val list = ArrayList(it)
-                setupListOfDateINtoRecycleVIew(list, NotesDao)
+        val spinner = findViewById<Spinner>(R.id.spinnerFilter)
+        val adapter = ArrayAdapter(
+            this,
+            android.R.layout.simple_spinner_item,
+            labelSuggestions
+        )
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        spinner.adapter = adapter
+        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+                val label = parent?.getItemAtPosition(position).toString()
+                lifecycleScope.launch {
+                    NotesDao.fetchAllNotes().collect {
+                        val list = ArrayList(it)
+                        if (label == "All") {
+                            setupListOfDateINtoRecycleVIew(list, NotesDao)
+                        }
+                        else {
+                            val finalList = ArrayList<NotesEntity>()
+                            for (NotesEntity in list) {
+                                val hash_set = HashSet<String>()
+                                hash_set.addAll(NotesEntity.Labels.split(", "))
+                                if (hash_set.contains(label)) {
+                                    finalList.add(NotesEntity)
+                                }
+
+                            }
+                            setupListOfDateINtoRecycleVIew(finalList, NotesDao)
+                        }
+                    }
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+                // Do nothing
             }
         }
+
+        //////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
What's this for?

Reply:
Nothing, I was having problem to find my code scrolling so made this, I will remove it